--- a/imxrt1062_t41.ld	2020-07-01 12:30:19.000000000 +0100
+++ b/imxrt1062_t41.ld	2021-09-15 16:13:11.769501800 +0100
@@ -85,6 +85,8 @@
 	_heap_start = ADDR(.bss.dma) + SIZEOF(.bss.dma);
 	_heap_end = ORIGIN(RAM) + LENGTH(RAM);
 
+	_extram_alloc = SIZEOF(.bss.extram) + 0x70000000;
+	
 	_itcm_block_count = (SIZEOF(.text.itcm) + SIZEOF(.ARM.exidx) + 0x7FFF) >> 15;
 	_flexram_bank_config = 0xAAAAAAAA | ((1 << (_itcm_block_count * 2)) - 1);
 	_estack = ORIGIN(DTCM) + ((16 - _itcm_block_count) << 15);
