--- a/imxrt1062_t41.ld	2022-07-29 10:07:35.796350479 +0100
+++ b/imxrt1062_t41.ld	2023-03-23 21:02:24.424210663 +0000
@@ -101,6 +101,8 @@
 	_extram_start = ADDR(.bss.extram);
 	_extram_end = ADDR(.bss.extram) + SIZEOF(.bss.extram);
 
+	_extram_alloc = SIZEOF(.bss.extram) + 0x70000000;
+	
 	_itcm_block_count = (SIZEOF(.text.itcm) + SIZEOF(.ARM.exidx) + 0x7FFF) >> 15;
 	_flexram_bank_config = 0xAAAAAAAA | ((1 << (_itcm_block_count * 2)) - 1);
 	_estack = ORIGIN(DTCM) + ((16 - _itcm_block_count) << 15);
